soccer_model_msgs/msg/Obstacle.msg

```diff
@@ -0,0 +1,5 @@
+# Describes a filtered obstacle from modelling modules, in 3d coordinates.
+# An Obstacle is defined as anything on the field that could not be classified as
+# one of the known types (such as Robot or Goalpost).
+
+soccer_geometry_msgs/PointWithCovariance point
```

Comment:
Of course, we don't know what an obstacle is, but couldn't it potentially also have a twist?

Reply:
I think it could be useful, but I don't see how useful it is in adding the field... (which was the reason I didn't have Twist for the Robot.msg originally too). If some team needs this field, we could add it I guess?

Reply:
I think it could be the case and useful if e.g. a kalman filter also models the velocities of a set of objects. But the objects currently feature only a position and no pose, but the twist also includes rotation information.

Reply:
I don't have a strong opinion, but personally I would use a pose and a twist for all of them. This makes also processing easier/more standardized as all objects have a similar msg layout and the few bytes that are wasted for most of the teams don't matter that much compared to other overheads. Also changing this in the future if it is needed adds breaking changes.

Reply:
So, I'm not too concerned about the size of the ros msg, but I am concerned about breaking api changes. Not that we should completely avoid them, but we should try and minimize them.

I see msgs as a "promise" made between all publisher and all subscribers. Publishers have a responsibilty to fill all the fields of the msg with valid information, and the subscriber has the right to assume valid information in the fields.

If the publisher is unable to fill a field (eg. can't fill the twist field because velocity isn't tracked), the subscribers need a way to know that the fields aren't set. (eg. leaving a twist at 0 would mean a 100% confidence twist of (0, 0, 0))

@Flova are you suggesting a pose for the Ball too?

Reply:
Right, we do something similar (super high default covariance) with the ball if it was not observed yet.

Regarding the ball I am okay with keeping it as it is, as one could estimate the rotational velocity without having an absolute rotation value, which is hard to determine due to the symmetries. But a pose would unify things which is also nice. So I don't have a real opinion on it.